index.js

```diff
@@ -1,88 +1,97 @@
-// ***** Question 1 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 1", "color: red") 
+let counter = 1
+    counter = 2
+console.log(counter) 
 
 
-// ***** Question 2 *****
-// *** Uncomment the lines below to test
-// *** after testing, comment the line causing the error back in 
-// *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
-// => "Raffy"
-// name = "Not Raffy" 
-// => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 2", "color: red") 
+const name = "simon"
+console.log(name)
 
 
 
-// ***** Question 3 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
+function drinkWater(currentThirstLevel){
+  console.log("Man I sure am thirsty")
+  currentThirstLevel --
+  console.log( "Ahh that hits the spot")
+  return(currentThirstLevel) 
+}
+console.log(drinkWater(12))
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
 
 
+console.log("%cQuestion 4", "color: red") 
 
-// ***** Question 4 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+function sameSameButDifferent (num, maybeNum){
+  if (num === maybeNum ){
+    return("same same")
+  }
+  else if (num == maybeNum){
+    return("same same (but different)")
+  }
+  else
+  {
+    return("different")
+  }
```

Comment:
- The key here was understanding the difference between `===` and `==`. You could also convert the string to a number with parseInt to avoid == :

`} else if (num === parseInt(maybeNum)) {`

- Only small thing - there's no need for the parentheses in the return on lines 30, 33, and 37.